scripts/genetic_backtester/darwin.js

```diff
@@ -39,8 +44,213 @@ let NEUTRAL_RATE_AUTO = false
 
 let iterationCount = 0
 
+let darwinMonitor = {
+  periodDurations: [],
+  phenotypes: [],
+
+  // numberOfPeriodsIn: function(so) {
+  //   // Adapted from sim.js logic to similarly figure out how much time is being processed
+  //   if (so.start) {
+  //     so.start = moment(so.start, 'YYYYMMDDhhmm')
+  //     if (so.days && !so.end) {
+  //       so.end = so.start.clone().add(so.days, 'days')
+  //     }
+  //   }
+  //   if (so.end) {
+  //     so.end = moment(so.end, 'YYYYMMDDhhmm')
+  //     if (so.days && !so.start) {
+  //       so.start = so.end.clone().subtract(so.days, 'days')
+  //     }
+  //   }
+  //   if (!so.start && so.days) {
+  //     so.start = moment().subtract(so.days, 'days')
+  //   }
+  //
+  //   // Figure out how many "periods" total this will be, then estimate time per period in order to estimate length of other jobs
+  //   if (so.start && so.end) {
+  //     var actualStart = moment( tb(so.start.valueOf()).resize(so.period_length).subtract(so.min_periods + 2).toMilliseconds() );
+  //     var diff = so.end.diff(actualStart);
+  //     var periods = tb(diff).resize(so.period_length);
+  //     return periods.value;
+  //   }
+  //   return 1;
+  // },
```

Comment:
Any reason for keepeing this outcommented code (and the other blocks) in the PR?

Reply:
This was part of an earlier attempt to measure time by first figuring out how many periods are being processed, and then measuring time per period. I quickly found out that time per period can vary based on lots of different things. However since that part worked, I was hesitant to get rid of it entirely in case someone else here had an idea of how to leverage it.